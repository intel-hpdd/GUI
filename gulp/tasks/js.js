//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
