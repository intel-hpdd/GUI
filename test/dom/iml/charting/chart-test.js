// @flow

import Inferno from 'inferno';

import { renderToSnapshot } from '../../../test-utils.js';

const Child = (props: Object) =>
  <div>
    {JSON.stringify(props[props.propName])}
  </div>;

describe('charting', () => {
  let Chart;

  beforeEach(() => {
    jest.mock('@iml/debounce', () => fn => fn);

    Chart = require('../../../../source/iml/charting/chart.js').default;
  });

  it('should match', () => {
    expect(
      renderToSnapshot(
        <Chart
          margins={{ top: 50, right: 50, bottom: 50, left: 50 }}
          points={[1, 2, 3]}
        >
          <Child propName="margins" />
          <Child propName="dimensions" />
          <Child propName="points" />
        </Chart>
      )
    ).toMatchSnapshot();
  });

  it('should set state on resize', () => {
    const root = document.createElement('div');
    root.style.setProperty('width', '500px');
    root.style.setProperty('height', '500px');

    Inferno.render(
      <Chart
        margins={{ top: 50, right: 50, bottom: 50, left: 50 }}
        points={[1, 2, 3]}
      >
        <Child propName="margins" />
        <Child propName="dimensions" />
        <Child propName="points" />
      </Chart>,
      root
    );

    // $FlowFixMe: Mocking for test
    root.querySelector('svg').getBoundingClientRect = () => ({
      width: 500,
      height: 500
    });

    window.dispatchEvent(new Event('resize', { bubbles: true }));

    expect(root).toMatchSnapshot();
  });

  it('should cleanup when unmounting', () => {
    jest.spyOn(window, 'removeEventListener');

    const root = document.createElement('div');
    root.style.setProperty('width', '500px');
    root.style.setProperty('height', '500px');

    Inferno.render(
      <Chart
        margins={{ top: 50, right: 50, bottom: 50, left: 50 }}
        points={[1, 2, 3]}
      >
        <Child propName="margins" />
        <Child propName="dimensions" />
        <Child propName="points" />
      </Chart>,
      root
    );

    Inferno.render(null, root);
  });
});
