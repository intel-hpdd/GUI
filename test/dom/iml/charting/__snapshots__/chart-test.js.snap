// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`charting should match 1`] = `
<div>
  <svg
    class="charting"
    style="width: 100%; height: 100%;"
  >
    <g
      class="charting-group"
      transform="translate(50,50)"
    >
      <div>
        {"top":50,"right":50,"bottom":50,"left":50}
      </div>
      <div>
        {"usableWidth":-100,"usableHeight":-100}
      </div>
      <div>
        [1,2,3]
      </div>
    </g>
  </svg>
</div>
`;

exports[`charting should set state on resize 1`] = `
<div
  style="width: 500px; height: 500px;"
>
  <svg
    class="charting"
    style="width: 100%; height: 100%;"
  >
    <g
      class="charting-group"
      transform="translate(50,50)"
    >
      <div>
        {"top":50,"right":50,"bottom":50,"left":50}
      </div>
      <div>
        {"usableWidth":400,"usableHeight":400}
      </div>
      <div>
        [1,2,3]
      </div>
    </g>
  </svg>
</div>
`;
